### PR TITLE
Update TF notebooks

### PR DESCRIPTION
**Pull request by Rocketknight1**
This PR will update TF notebooks with the following main changes, as well as any other fixes I find while I'm editing them:

- Use XLA compilation when suitable, add an explanation about XLA to all notebooks
- Replace `to_tf_dataset` with `model.prepare_tf_dataset`, which has smart column inference and is much more concise.
- Add inference sections to show how to use the trained models to get predictions (cc @NielsRogge and discussion [here](https://huggingface.slack.com/archives/C01N44FJDHT/p1658238421669899))

Right now the PR just has the text classification notebook so that I can get feedback on the inference section, but once people are happy with it I'll edit and push the other notebooks too!

### CODE REVIEW

File: examples/text_classification-tf.ipynb [hunk 1]
```diff
@@ -84,7 +84,7 @@
      "name": "stdout",
```
Comment (sgugger) [inline examples/text_classification-tf.ipynb:84]: <pre class="ql-syntax" spellcheck="false">Line #3.    pipe = pipeline("text-classification", "Rocketknight1/distilbert-base-uncased-finetuned-cola", framework="tf")
</pre><p>Let's call this <code>classifier</code> or something along those lines. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/notebooks/pull/207/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='74'/>

**Review by sgugger — APPROVED** [review]
Nice, thanks for updating those!
You could also switch to `evaluate` for metric computation while you're at it.

**Comment by Rocketknight1** [comment]
I just tried, but there is a very perverse bug here: `evaluate` loads `modeling_utils.py`, but because we have a folder called `accelerate` in the notebooks folder, `is_accelerate_available` returns `True` when we run it in any of these notebooks, and then we get a crash when `modeling_utils.py` tries to import from it.

@sgugger What's your preference for fixing this? I can rename the folder, or should we change how `is_accelerate_available` works?

**Comment by sgugger** [comment]
Mmm, maybe it's time to move the examples to examples/transformers so there is space for other libs, but it will break all links...
As a quick fix, maybe rename accelerate to `accelerate_examples` which will break less links?

**Comment by Rocketknight1** [comment]
Sure!

**Review by gante — APPROVED** [review]
👍  LGTM

**Comment by Rocketknight1** [comment]
@sgugger @gante This PR is now ready, with all notebooks updated! @gante in particular, please take a look at the summarization and translation notebooks - this is where I explained and used XLA generation in both the `KerasMetricCallback` as well as the Inference section. Performance was superb, but please make sure I'm following the best practices!

**Comment by Rocketknight1** [comment]
Going to merge this for the release now - I've run all the notebooks and they work! If people have nits we can just fix them after the release.
